tests/integrations/sanic/test_sanic.py

```diff
@@ -284,3 +296,110 @@ async def runner():
+        integration_args,
+        url,
+        expected_status,
+        expected_transaction_name,
+        expected_source=None,
+    ):
+        # type: (Iterable[Optional[Container[int]]], str, int, Optional[str], Optional[str]) -> None
+        """
+        expected_transaction_name of None indicates we expect to not receive a transaction
+        """
+        self.integration_args = integration_args
+        self.url = url
+        self.expected_status = expected_status
+        self.expected_transaction_name = expected_transaction_name
+        self.expected_source = expected_source
+
+
+@pytest.mark.skipif(
+    not PERFORMANCE_SUPPORTED, reason="Performance not supported on this Sanic version"
+)
+@pytest.mark.parametrize(
+    "test_config",
+    [
+        TransactionTestConfig(
+            # Transaction for successful page load
+            integration_args=(),
+            url="/message",
+            expected_status=200,
+            expected_transaction_name="hi",
+            expected_source=TRANSACTION_SOURCE_COMPONENT,
+        ),
+        TransactionTestConfig(
+            # Transaction still recorded when we have an internal server error
+            integration_args=(),
+            url="/500",
+            expected_status=500,
+            expected_transaction_name="fivehundred",
+            expected_source=TRANSACTION_SOURCE_COMPONENT,
+        ),
+        TransactionTestConfig(
+            # By default, no transaction when we have a 404 error
+            integration_args=(),
+            url="/404",
+            expected_status=404,
+            expected_transaction_name=None,
+        ),
+        TransactionTestConfig(
+            # With no ignored HTTP statuses, we should get transactions for 404 errors
+            integration_args=(None,),
+            url="/404",
+            expected_status=404,
+            expected_transaction_name="/404",
+            expected_source=TRANSACTION_SOURCE_URL,
+        ),
+        TransactionTestConfig(
+            # Transaction can be suppressed for other HTTP statuses, too, by passing config to the integration
+            integration_args=({200},),
+            url="/message",
+            expected_status=200,
+            expected_transaction_name=None,
+        ),
+    ],
+)
+def test_transactions(test_config, sentry_init, app, capture_events):
+    # type: (TransactionTestConfig, Any, Any, Any) -> None
+
+    # Init the SanicIntegration with the desired arguments
+    sentry_init(
+        integrations=[SanicIntegration(*test_config.integration_args)],
+        traces_sample_rate=1.0,
+    )
+    events = capture_events()
+
+    # Make request to the desired URL
+    _, response = app.test_client.get(test_config.url)
+    assert response.status == test_config.expected_status
+
+    # Extract the transaction event by inspecting the event types
+    transaction_event = next(
+        (e for e in events if "type" in e and e["type"] == "transaction"), None
+    )
```

Comment:
Since we're expecting either exactly one or zero transactions, we could assert here that we never get two or more transaction events. Right now we're just taking the first transaction, but there's no telling whether there might be another one after that, which would be a bug.